### PR TITLE
CO-2925 confidential identities no cert

### PR DESCRIPTION
**Pull request by willhr3**
- Stripped out some of the certificates dependencies from confidential identities
- Created a flow for requesting a new public key 
- Keys and Parties are stored in SignedPublicKey map 

Still to do:
- Share keys flow
- Sync keys flow

### CODE REVIEW

**Review by roger-that-dev — CHANGES_REQUESTED** [review]
Looks good so far - definitely going in the right direction I think! Added some points around the request key flows.

Some other bits we'll need to make sure we have addressed:
1. This PR is based upon @roastario 's initiate flow with abstract party branch. We'll want to merge this is before any of this work.
2. The identity service changes are also in this PR, I think we should split them out into a separate PR for ease of review. We should also get that PR merged before any of the work on confidential identities is merged in.
3. In the identity service we'll need to check that migration from the old certificates based identity service to the new simpler service doesn't lose any of the mappings and properly migrates the existing mappings over to the new approach. This is always more difficult to get right than you first think.
4. With all of this stuff we'll need to update the documentation and add kDocs to all of the new APIs. For now don't worry about this as it's just a PoC.
5. We'll need to update the api-current.txt file as we've made some API changes to Corda in this PR (otherwise the CI checks won't pass).

Next steps: once we can fully demonstrate this working then we can set about putting together a design doc for a design review board, with the lessons learnt from building this.

Good work - cheers!

File: confidential-identities/build.gradle [hunk 1]
```diff
@@ -11,7 +11,7 @@ description 'Corda Experimental Confidential Identities'
 
 dependencies {
     cordaCompile project(':core')
-
+    cordaCompile (group: 'net.corda', name: 'corda-node', version: '5.0-SNAPSHOT')
```
[the remark this answers is not in the document]
Comment (willhr3) [reply confidential-identities/build.gradle]: We end up with a circular dependency when you use that which is strange.

File: node/src/main/kotlin/net/corda/node/services/identity/PersistentIdentityService.kt [hunk 2]
```diff
@@ -34,10 +38,11 @@ class PersistentIdentityService(cacheFactory: NamedCacheFactory) : SingletonSeri
     companion object {
         private val log = contextLogger()
 
-        fun createPKMap(cacheFactory: NamedCacheFactory): AppendOnlyPersistentMap<SecureHash, PartyAndCertificate, PersistentIdentity, String> {
+
+        fun createKeyToPartyAndCertMap(cacheFactory: NamedCacheFactory): AppendOnlyPersistentMap<SecureHash, PartyAndCertificate, PersistentIdentityCert, String> {
             return AppendOnlyPersistentMap(
```
Comment (roger3cev) [inline node/src/main/kotlin/net/corda/node/services/identity/PersistentIdentityService.kt]: would be ncie to see if we can remove all the cert stuff, then deprecate all the cert based methods. We'll also need to deal with migration from the cert based mapping to the non cert based mapping.

**Comment by roger3cev** [comment]
Let's close this PR now. Cheers!

**Pull request closed without merging**